### ISSUE TITLE
getChartDataRequest handling does not always account for async query responses

### ISSUE THREAD

**Issue by benjreinhart** [apache/superset#15183]
[`getChartDataRequest`](https://github.com/apache/superset/blob/7dc0cee5be33848f46e443fcf3181b605811945b/superset-frontend/src/chart/chartAction.js#L202) is currently used to grab data for charts in 6 places. However, three of those call sites do not account for the (different) response returned when the Global Async Queries feature is enabled.

Here are the call sites (the checked ones appear to properly handle GAQ):

- [X] [src/chart/chartAction.js#L392](https://github.com/apache/superset/blob/7dc0cee5be33848f46e443fcf3181b605811945b/superset-frontend/src/chart/chartAction.js#L392)
- [ ] [src/chart/chartAction.js#L543](https://github.com/apache/superset/blob/7dc0cee5be33848f46e443fcf3181b605811945b/superset-frontend/src/chart/chartAction.js#L543)
- [X] [src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterValue.tsx#L112](https://github.com/apache/superset/blob/7dc0cee5be33848f46e443fcf3181b605811945b/superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterValue.tsx#L112)
- [X] [src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx#L368](https://github.com/apache/superset/blob/7dc0cee5be33848f46e443fcf3181b605811945b/superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx#L368)
- [ ] [src/explore/components/controls/ViewQueryModal.tsx#L56](https://github.com/apache/superset/blob/7dc0cee5be33848f46e443fcf3181b605811945b/superset-frontend/src/explore/components/controls/ViewQueryModal.tsx#L56)
- [ ] [src/explore/components/DataTablesPane/index.tsx#L146](https://github.com/apache/superset/blob/7dc0cee5be33848f46e443fcf3181b605811945b/superset-frontend/src/explore/components/DataTablesPane/index.tsx#L146)

### Expected results

All call sites should be able to properly handle the async queries experience when the feature is enabled.

### Actual results

Some actions, such as clicking "Run in SQL Lab," will not work properly when GAQ is enabled.

### Additional context

Once https://github.com/apache/superset/pull/15157 is merged, all these call sites will need to additionally make sure they handle the synchronous response in addition to asynchronous response when GAQ is enabled.

**Comment by rusackas** [comment]
Not sure if this is still an issue, but we'll close this as stale since it's been silent for so long, and we're trying to steer toward a more actionable Issues backlog. If people are still encountering this in current versions (currently 3.x) please open a new Issue or a PR to address the problem.
